catalog/rest.go

```diff
@@ -491,11 +485,15 @@ func (r *RestCatalog) fetchAccessToken(cl *http.Client, creds string, opts *opti
 		clientID, clientSecret = "", clientID
 	}
 
+	scope := "catalog"
+	if opts.scope != nil {
+		scope = *opts.scope
+	}
```

Comment:
if it's not a pointer we can just do `if opts.scope != ""` and avoid the heap

Reply:
👍